src/transit.js

```diff
@@ -115,10 +115,12 @@ class Transit {
 					this.logger.warn("Connection is failed.", err.message);
 					this.logger.debug(err);
 
-					setTimeout(() => {
-						this.logger.info("Reconnecting...");
-						doConnect();
-					}, 5 * 1000);
+					if (this.tx.opts && this.tx.opts.protocolId !== "MQTT") {
```

Comment:
I think it doesn't filter out that there is one or multiple servers configured, right?

Reply:
I'm not sure I understood you and if not please explain to me again.

You are right, it only filter out by protocol.
The re-connection is not an issue when there is one server configured. 
On cases of multiple servers configured there could be unnecessary re-connection attempts.

However, since the mqtt lib already handles the re-connection mechanism we can comment out the re-connection from here for MQTT protocol only (regardless of number of servers). 

Therefore it doesn't matter how many servers are configured, if we are dealing with MQTT we should leave the re-connection to the mqtt lib.

Reply:
A flag on the `tx.opts` (like `noReconnect`) object would be better suited since this could end up being applied to other protocols later on. The transporter constructor could set a default for this but also allow for it to be passed in as this would allow for more customization when creating your own transporters.